src/dev/getting_started.md

```diff
@@ -83,6 +83,65 @@ required) for this directory instead of the default Rust compiler.
 Then you can build and test the `yk` repo using `cargo xtask` commands, for
 example `cargo xtask test`.
 
+## Rust Analyzer
+
+[Rust-analyzer](https://github.com/rust-analyzer/rust-analyzer) (RA hencforth)
+is a language server implementation for Rust.
+
+Due to [the workspace model of the `yk` repo](../tech/yk_structure.md) and the
+`x.py` build system in `ykrustc`, RA won't work out of the box. Configuration
+is dependent on the LSP plugin used, but in general:
+
+ - For the `ykrustc` repo, you need to tell RA to run `x.py` and not `cargo`.
+   This is done via the `rust-analyzer.checkOnSave.overrideCommand` option.
+
+ - For the `yk` repo, you need to point RA at the extra internal workspace via
+the `rust-analyzer.linkedProjects` option.
+
+For example, for [vim-lsp](https://github.com/prabirshrestha/vim-lsp) you would
+use a configuration similar to the following:
+
+```
+if executable('rust-analyzer')
+    if filereadable("x.py")
```

Comment:
I think this is fine for the time being, but *probably* this path should be relative to the file loaded or something? I don't know how vim handles that and I don't want to hurt my eyes by trying to find out :p

Reply:
Yeah. I couldn't see a way to do in with vim-lsp.

Reply:
Let's not worry about it then.